core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -215,21 +215,17 @@ private void addAndWaitIfFull(CompletableFuture future, List<CompletableFuture>
    }
 
    @Override
-   public CompletableFuture<Boolean> entryExpiredInMemoryFromIteration(InternalCacheEntry<K, V> entry, long currentTime) {
+   public boolean entryExpiredInMemoryFromIteration(InternalCacheEntry<K, V> entry, long currentTime) {
       // We need to synchronize on the entry since {@link InternalCacheEntry} locks the entry when doing an update
       // so we can see both the new value and the metadata
       boolean expiredMortal;
       synchronized (entry) {
          expiredMortal = ExpiryHelper.isExpiredMortal(entry.getLifespan(), entry.getCreated(), currentTime);
       }
-      if (expiredMortal) {
-         // Lifespan was expired - but we don't want to take the hit of causing an expire command to be fired
-         return CompletableFutures.completedTrue();
-      } else {
-         // Max idle expiration - we just return it (otherwise we would have to incur remote overhead)
-         // This entry will be removed on next get or reaper running
-         return CompletableFutures.completedFalse();
-      }
+      // We only report an entry as being expired if it was mortal - transient (maxIdle) could be a false positive
+      // as another node may have had a more recent read - so we just include it with the assumption the entry
+      // will be cleaned up on next get or reaper running
```

Comment:
I always forget why this is not a problem with lifespan expiration, because the create timestamp is not the same on all the nodes either. Admittedly the variation is smaller for the create timestamp than for the last access timestamp.

Reply:
This is not a problem for lifespan, because it is coded that way. If any node finds it is expired via lifespan it removes it everywhere. We don't verify it is expired on all, if one node is close enough to being expired, that is enough reason to remove it, when it is via lifespan.

Reply:
Could you reword the comment to include that explanation?

Reply:
Well the code isn't removing the lifespan expired entry here, has to wait until a get or reaper as it mentions. I can try to be more explicit I guess.

Reply:
Added new comment - maybe it is more clear now?

Reply:
Yes, thanks!